docs/core/seeding.rst

```diff
@@ -194,46 +228,55 @@ scattering for the triplet as well.
 circle. Therefore, the deviation from a straight line is compared to the
 maximum allowed scattering at minimum pT scaled by the forward angle (as
-calculated above). Both the check against min pT as the check against the
+calculated above). Both the check against min pT and the check against the
 calculated pT (discussed further below) take the correlated measurement
 uncertainty into account.
 
 .. code-block:: cpp
 
-   // add errors of spB-spM and spM-spT pairs and add the correlation term
-   // for errors on spM
-   float error2 = lt.Er + ErB +
-                  2 * (cotThetaB * lt.cotTheta * varianceRM + varianceZM) *
-                      iDeltaRB * lt.iDeltaR;
-
-   float deltaCotTheta = cotThetaB - lt.cotTheta;
-   float deltaCotTheta2 = deltaCotTheta * deltaCotTheta;
-   float error;
-   float dCotThetaMinusError2;
-   // if the error is larger than the difference in theta, no need to
-   // compare with scattering
-   if (deltaCotTheta2 - error2 > 0) {
-     deltaCotTheta = std::abs(deltaCotTheta);
-     // if deltaTheta larger than the scattering for the lower pT cut, skip
-     error = std::sqrt(error2);
-     dCotThetaMinusError2 =
-         deltaCotTheta2 + error2 - 2 * deltaCotTheta * error;
-     // avoid taking root of scatteringInRegion
-     // if left side of ">" is positive, both sides of unequality can be
-     // squared
-     // (scattering is always positive)
-
-     if (dCotThetaMinusError2 > scatteringInRegion2) {
-       continue;
-     }
-   }
+	// add errors of spB-spM and spM-spT pairs and add the correlation term
+	// for errors on spM
+	float error2 = lt.Er + ErB +
+								 2 * (cotThetaAvg2 * varianceRM + varianceZM) * iDeltaRB *
+										 lt.iDeltaR;
+
+	float deltaCotTheta = cotThetaB - cotThetaT;
+	float deltaCotTheta2 = deltaCotTheta * deltaCotTheta;
+	// Apply a cut on the compatibility between the r-z slope of the two
+	// seed segments. This is done by comparing the squared difference
+	// between slopes, and comparing to the squared uncertainty in this
+	// difference - we keep a seed if the difference is compatible within
+	// the assumed uncertainties. The uncertainties get contribution from
+	// the  space-point-related squared error (error2) and a scattering term
+	// calculated assuming the minimum pt we expect to reconstruct
+	// (scatteringInRegion2). This assumes gaussian error propagation which
+	// allows just adding the two errors if they are uncorrelated (which is
+	// fair for scattering and measurement uncertainties)
+	if (deltaCotTheta2 > (error2 + scatteringInRegion2)) {
+		// additional cut to skip top SPs when producing triplets
+		if (m_config.skipPreviousTopSP) {
+			// break if cotTheta from bottom SP < cotTheta from top SP because
+			// the SP are sorted by cotTheta
+			if (cotThetaB - cotThetaT < 0) {
+				break;
+			}
+			t0 = t + 1;
+		}
+		continue;
+	}
 
 Now the check for scattering with calculated particle momentum. Momentum is
```

Comment:
Next/Following check takes into account estimate particle momentum (smaller scattering angle is permitted for higher momentum) and pseudorapidity (larger scattering takes into account amount of the material crosses that takes depends on the angle). (If that is the reason actually.)